src/utils/eventManager.ts

```
@@ -0,0 +1,122 @@
+    detail?: string
+    life?: number
+    closable?: boolean
+  }
+}
+
+/**
+ * Type-safe event manager for medical calculator events
+ * Provides automatic cleanup and proper TypeScript support
+ */
+export class TypeSafeEventManager {
+  private listeners = new Map<keyof MedicalCalculatorEventMap, Set<Function>>()
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)